apps/challenges/challenge_config_utils.py

```diff
@@ -220,436 +239,757 @@ def get_value_from_field(data, base_location, field_name):
+                    error = True
+                if "schema" not in data:
+                    message = self.error_messages_dict.get(
+                        "missing_leaderboard_schema"
+                    ).format(data.get("id"))
+                    self.error_messages.append(message)
+                    error = True
                 else:
-                    if current_leaderboard_config_ids and int(data["id"]) not in current_leaderboard_config_ids:
-                        error_messages.append("ERROR: Leaderboard {} doesn't exist. Addition of new leaderboard after challenge creation is not allowed.".format(data["id"]))
-                    leaderboard_ids.append(data["id"])
-    else:
-        message = "ERROR: There is no key leaderboard in the YAML file."
-        error_messages.append(message)
+                    if "labels" not in data["schema"]:
+                        message = self.error_messages_dict.get("missing_labels").format(
+                            data.get("id")
+                        )
+                        self.error_messages.append(message)
+                        error = True
+                    if "default_order_by" not in data["schema"]:
+                        message = self.error_messages_dict.get(
+                            "missing_default_order_by"
+                        ).format(data.get("id"))
+                        self.error_messages.append(message)
+                        error = True
+                    else:
+                        default_order_by = data["schema"]["default_order_by"]
+                        if (
+                            "labels" in data["schema"]
+                            and default_order_by not in data["schema"]["labels"]
+                        ):
+                            message = self.error_messages_dict.get(
+                                "wrong_default_order_by"
+                            ).format(default_order_by, data.get("id"))
+                            self.error_messages.append(message)
+                            error = True
+                if not error:
+                    serializer = LeaderboardSerializer(
+                        data=data, context={"config_id": data["id"]}
+                    )
+                    if not serializer.is_valid():
+                        serializer_error = str(serializer.errors)
+                        message = self.error_messages_dict.get(
+                            "leaderboard_schema_error"
+                        ).format(data["id"], serializer_error)
+                        self.error_messages.append(message)
+                    else:
+                        if (
+                            current_leaderboard_config_ids
+                            and int(data["id"]) not in current_leaderboard_config_ids
+                        ):
+                            message = self.error_messages_dict.get(
+                                "leaderboard_schema_error"
```

Comment:
This is very incorrect.
You are essentially removing an improvement here.

The current leaderboard config IDs check that new leaderboards cannot be added and old leaderboards cannot be deleted.
Please make sure you are not removing any of the previous checks.
`leaderboard_schema_error` is a generic error for any issues with schema. It does not check that.
```
 if current_leaderboard_config_ids and int(data["id"]) not in current_leaderboard_config_ids:
                        error_messages.append("ERROR: Leaderboard {} doesn't exist. Addition of new leaderboard after challenge creation is not allowed.".format(data["id"]))
                    leaderboard_ids.append(data["id"])
```

```
    for current_leaderboard_id in current_leaderboard_config_ids:
        if current_leaderboard_id not in leaderboard_ids:
            error_messages.append("ERROR: Leaderboard {} not found in config. Deletion of existing leaderboard after challenge creation is not allowed.".format(current_leaderboard_id))
```